### ISSUE TITLE
[question] Range mode timezone behavior differs depending on whether dates are included in the initial model array

### ISSUE THREAD

**Issue by TimJohns** [Vuepic/vue-datepicker#756]
I'm framing this Issue as a bug report, but I'm not convinced yet that it's a bug. It's at least a question/confusion, and may be a feature request for some additional example or documentation. I'm going ahead and opening the issue while I continue to experiment, in the hope that others who may be working with a similar use case may be able to either help me clarify the expected behavior or benefit from these observations. I realize a GitHub issue might not be the most appropriate forum for this discussion, so if there is a better place to discuss, please point me at it.

And always, thank you for all of the hard work you have put into this component - I really appreciate it.
 
**Describe the bug**
When the VueDatePicker is configured in range mode, the timezone behavior differs depending on whether or not there are existing dates provided in the v-model.

**To Reproduce**
1. Configure a `VueDatePicker` component with `range`, a `timezone` config, a 'start-time', and **an empty ref** for `v-model`.
2. Pick any two dates
3. Notice the dates emitted, particularly the offsets
4. Configure a `VueDatePicker` component with `range`, a `timezone` config, a 'start-time',  and **an array containing an existing start and end date** in the `v-model`.
5. Pick the same to dates as in step 2
6. Notice the dates emitted differ by offset from those emitted in Step 3

**Expected behavior**
The dates emitted by `update:model-value` would have the same offset regardless of whether initial values were supplied in the `v-model`. I'm using the date picker to select days in different time zones, and I was experimenting to make sure I understood the new timezone configuration behavior when I noticed this behavior. Note also that the values emitted by `update:model-timezone-value` appear to have this same behavior. I suspect this is some interaction with the `start-time` property - I noticed that it does not seem to be applied when there are initial dates provided.

### Example Component
```
<template>
  <VueDatePicker 
      v-model="dateRange"
      range
      :start-time="startTime"
      :timezone="timezone"
      :enable-time-picker=false
      :auto-apply=true
      @update:model-value="onUpdateRange"
      @update:model-timezone-value="onUpdateRangeTz"
    >
    <template #trigger>
      <button>Pick dates</button>
    </template>
  </VueDatePicker>
</template>

<script setup lang="ts">
import { ref } from 'vue';

import VueDatePicker from '@vuepic/vue-datepicker';
import '@vuepic/vue-datepicker/dist/main.css'

const dateRange = ref<Array<Date>>();
// const dateRange = ref<Array<Date>>([ new Date("2024-02-16T00:00:00.000Z"), new Date("2024-02-16T00:00:00.000Z") ]);
const startTime = ref([{ hours: 0, minutes: 0 }, { hours: 0, minutes: 0 }]);
const timezone = ref({
  timezone: "UTC",
  exactMatch: true,
  dateInTz: "UTC",
  emitTimezone: "UTC"
});

function onUpdateRange(dateRange: Array<Date>) {
  dateRange.forEach((date, index) => { console.log(`onUpdateRange ${index} `, date.toISOString())})
};

function onUpdateRangeTz(dateRange: Array<Date>) {
  dateRange.forEach((date, index) => { console.log(`onUpdateRangeTz ${index} `, date.toISOString())})
};
</script>
```
---
### Example Vitest/Vue Testing-Library test case

```
import { expect, describe, it, vi } from "vitest"
import '@testing-library/jest-dom/vitest'
import { fireEvent, render } from "@testing-library/vue";
import { flushPromises } from "@vue/test-utils";
import { defineComponent } from "vue";
import VueDatePicker from '@vuepic/vue-datepicker';


describe('VueDatePicker', () => {

  it('should update the dates', async () => {
//    const dateRange = [];
    const dateRange = [ new Date("2024-02-16T00:00:00.000Z"), new Date("2024-02-17T00:00:00.000Z") ];
//    const dateRange = [ "2024-02-16T00:00:00.000Z", "2024-02-17T00:00:00.000Z" ];
//    const dateRange = [ "2024-02-16T08:00:00.000Z", "2024-02-17T08:00:00.000Z" ];
//    const dateRange = [ new Date("2024-02-16T00:00:00.000Z"), new Date("2024-02-17T00:00:00.000Z") ];

    const onUpdateRange = vi.fn();
    const onUpdateRangeTz = vi.fn();
    
    const wrappedComponent = defineComponent({
      template: 
      `<VueDatePicker 
          v-model="dateRange"
          range
          :start-time="startTime"
          :timezone="timezone"
          :enable-time-picker=false
          :auto-apply=true
          @update:modelValue="onUpdateRange"
          @update:model-timezone-value="onUpdateRangeTz"      
            >
        <template #trigger>
          <button>Pick dates</button>
        </template>
      </VueDatePicker>
      `,
      components: { VueDatePicker },
      data: () => ({
        dateRange,
        startTime: [{ hours: 0, minutes: 0 }, { hours: 0, minutes: 0 }],
        timezone: {
          timezone: "UTC",
          exactMatch: true,
          dateInTz: "UTC",
          emitTimezone: "UTC"
        },
        onUpdateRange,
        onUpdateRangeTz
      }) 
    });

    const renderResult = await render(wrappedComponent);

    await fireEvent.click(await renderResult.getByRole('button', {name: 'Pick dates'})); // Open the datepicker
    await fireEvent.click(await renderResult.getByRole('button', {name: "Open years overlay"})); // Open the year picker
    await fireEvent.click(await renderResult.getByRole('gridcell', {name: '2024'})); // select 2024
    await fireEvent.click(await renderResult.getByRole('button', {name: "Open months overlay"})); // Open the month picker
    await fireEvent.click(await renderResult.getByRole('gridcell', {name: 'Feb'})); // select February

    await flushPromises();

    await fireEvent.click(await renderResult.getByText('1', {selector: ".dp__cell_inner.dp--past"})); // select the 1st      
    await fireEvent.click(await renderResult.getByText('2', {selector: ".dp__cell_inner.dp--past"})); // select the 2nd

    expect(onUpdateRange).toHaveBeenCalledWith(
      expect.arrayContaining([
        new Date("2024-02-01T08:00:00.000Z"),
        new Date("2024-02-02T08:00:00.000Z")
      ])
    );

    expect(onUpdateRangeTz).toHaveBeenCalledWith(
      expect.arrayContaining([
        new Date("2024-02-01T16:00:00.000Z"),
        new Date("2024-02-02T16:00:00.000Z")
      ])
    )

  });
});

```

**Comment by Jasenkoo** [comment]
The `timezone` prop will not convert the initial `v-model` value. It will use the date, as is. If you do not provide the `v-model`, the hours will be adjusted to match the timezone.

**Comment by TimJohns** [comment]
Following up for anyone who sees this later: **I had omitted setting the `utc` property to `preserve`**. With `utc="preserve"`, I have confirmed the behavior is consistent for both cases (where the `v-model` contains initial date range values as well as when it is empty), which totally makes sense now.

Note, however, that with `utc` set to `preserve`, the model and corresponding updates are as IsoStrings rather than dates. I will enter a separate GitHub issue for that topic - that's actually desirable in my case (since I was converting the dates to IsoStrings anyway) but may not be intended, and may be worth documenting in either case.

Thanks again.
